packages/fe/src/app/components/core/form/widgets/DataSelectWidget.tsx

```diff
@@ -0,0 +1,164 @@
+import { ChangeEvent, FocusEvent } from 'react';
+import {
+  ariaDescribedByIds,
+  labelValue,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+import {MenuItem, TextField, TextFieldProps} from "@mui/material";
+import {types} from "@app/shared/types";
+import {isQueryableStateListDescription, QueryableStateListDescription} from "@event-engine/descriptions/descriptions";
+import {useApiQuery} from "@frontend/queries/use-api-query";
+import * as jexl from "jexl";
+import AddBrand from "@frontend/app/components/fleet-management/commands/AddBrand";
+import {commands} from "@frontend/app/components/commands";
+
+// Copied from: https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/material-ui/src/SelectWidget/SelectWidget.tsx
+// and modified to use useApiQuery and turn result into select options
+
+interface ParsedUiOptions {
+  data: QueryableStateListDescription,
+  label: string,
+  value: string,
+  addItemCommand: string | null,
+}
+
+const parseOptions = (options: any): ParsedUiOptions => {
+
+  if(!options.data || typeof options.data !== "string") {
+    throw new Error('DataSelect: no "data" attribute configured!');
+  }
+
+  if(!types[options.data]) {
+    throw new Error(`DataSelect: Unknown type "${options.data}"`);
+  }
+
+  const vo = types[options.data];
+
+  if(!isQueryableStateListDescription(vo.desc)) {
+    throw new Error(`DataSelect: Type "${options.data}" is not a queryable list`);
+  }
+
+  if(!options.label || typeof options.label !== "string") {
+    throw new Error(`DataSelect: ui:options "label" is not a string`);
+  }
+
+  if(!options.value || typeof options.value !== "string") {
+    throw new Error(`DataSelect: ui:options "value" is not a string`);
+  }
+
+  if(options.addItemCommand && typeof options.addItemCommand !== "string") {
+    throw new Error(`DataSelect: ui:options "addItemCommand" is not a valid command name`)
+  }
+
+  return {
+    data: vo.desc,
+    label: options.label,
+    value: options.value,
+    addItemCommand: options.addItemCommand || null,
+  }
+}
+
+export default function DataSelectWidget<
+  T = any,
+  S extends StrictRJSFSchema = RJSFSchema,
+  F extends FormContextType = any
+  >({
+      schema,
+      id,
+      options,
+      label,
+      hideLabel,
+      required,
+      disabled,
+      readonly,
+      placeholder,
+      value,
+      multiple,
+      autofocus,
+      onChange,
+      onBlur,
+      onFocus,
+      rawErrors = [],
+      registry,
+      uiSchema,
+      hideError,
+      formContext,
+      ...textFieldProps
+    }: WidgetProps<T, S, F>) {
+
+  const selectOptions: {label: string, value: string, readonly: boolean}[] = [];
+  const parsedOptions = parseOptions(options);
+
+  const query = useApiQuery(parsedOptions.data.query, {});
+
+  if(query.isSuccess) {
+    (query.data as any[]).forEach(item => {
+      selectOptions.push({
+        label: jexl.evalSync(parsedOptions.label, {data: item}),
```

Comment:
[Jexl](https://github.com/TomFrost/Jexl) is used in many places across Cody Engine to give prooph board users a scoped possibility for dynamic logic. Here one can define how each state list item (referenced as "data" within the Jexl scope) is turned into label and value for the selectbox option.

_Add Model Metadata Example again_
```json
"ui:options": {
        "data": "FleetManagement.Car.BrandList",
        "label": "data.name",
        "value": "data.brandId",
        "addItemCommand": "FleetManagement.AddBrand"
}
```